wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -482,6 +496,45 @@ private RecipeParser getRecipeParser(StageContext context)
     return new GrammarBasedParser(context.getNamespace(), new MigrateToV2(directives).migrate(), registry);
   }
 
+  /**
+   * This method emits all metrics for the given list of directives
+   * @param directives a list of Wrangler directives
+   * @param metrics CDAP {@link Metrics} object using which metrics can be emitted
+   * @throws DirectiveLoadException
+   */
+  private void emitDirectiveMetrics(List<Directive> directives, Metrics metrics) throws DirectiveLoadException {
+    for (Directive directive : directives) {
+      List<WranglerMetric> wranglerMetrics = new ArrayList<>();
+
+      // Emit directive usage metric only if directive is present in system directive registry
+      if (registry.get(Contexts.SYSTEM, directive.define().getDirectiveName()) != null) {
+        wranglerMetrics.add(getDirectiveUsageMetric(directive.define().getDirectiveName()));
+      }
+
+      if (directive instanceof EntityWithMetrics) {
+        wranglerMetrics.addAll(((EntityWithMetrics) directive).getMetrics());
+      }
+
+      for (WranglerMetric wranglerMetric : wranglerMetrics) {
+        Metrics child = metrics.child(wranglerMetric.getTags());
+        if (wranglerMetric.getType().equals(MetricType.COUNTER)) {
```

Comment:
we should use switch case for readability

Reply:
Done